Sources/LLVM/StructType.swift

```diff
@@ -57,15 +57,33 @@ public struct StructType: IRType {
   /// Creates a constant value of this structure type initialized with the given
   /// list of values.
   ///
+  /// - precondition: values.count == aggregate.elementCount
+  /// - parameter values: A list of values of members of this structure.
+  ///
+  /// - returns: A value representing a constant value of this structure type.
+  public func constant(values: [IRValue]) -> Constant<Struct> {
+    assert(numericCast(values.count) == LLVMCountStructElementTypes(llvm), "The number of values must match the number of elements in the aggregate")
+    var vals = values.map { $0.asLLVM() as Optional }
+    return vals.withUnsafeMutableBufferPointer { buf in
+      return Constant(llvm: LLVMConstNamedStruct(asLLVM(),
+                                                 buf.baseAddress,
+                                                 UInt32(buf.count)))
+    }
+  }
+
+  /// Creates a complex constant value initialized with the given list of values.
+  ///
   /// - parameter values: A list of values of members of this structure.
   /// - parameter isPacked: Whether or not this structure is 1-byte aligned with
   ///   no packing between fields.  Defaults to `false`.
   ///
-  /// - returns: A value representing a constant value of this structure type.
-  public static func constant(values: [IRValue], isPacked: Bool = false) -> IRValue {
+  /// - returns: A value representing a complex constant value with given the values.
```

Comment:
Nit: "complex" is confusable for things like complex numbers in this context.  "Aggregate" or "Structure" works better.